src/pt/randomscan/src/eu/kanade/tachiyomi/extension/pt/randomscan/LuraToon.kt

```diff
@@ -3,61 +3,191 @@ package eu.kanade.tachiyomi.extension.pt.randomscan
 import android.app.Application
 import android.content.SharedPreferences
 import androidx.preference.PreferenceScreen
+import eu.kanade.tachiyomi.extension.pt.randomscan.dto.Capitulo
+import eu.kanade.tachiyomi.extension.pt.randomscan.dto.CapituloPagina
+import eu.kanade.tachiyomi.extension.pt.randomscan.dto.MainPage
+import eu.kanade.tachiyomi.extension.pt.randomscan.dto.Manga
+import eu.kanade.tachiyomi.extension.pt.randomscan.dto.SearchResponse
 import eu.kanade.tachiyomi.lib.randomua.addRandomUAPreferenceToScreen
 import eu.kanade.tachiyomi.lib.randomua.getPrefCustomUA
 import eu.kanade.tachiyomi.lib.randomua.getPrefUAType
 import eu.kanade.tachiyomi.lib.randomua.setRandomUserAgent
-import eu.kanade.tachiyomi.multisrc.peachscan.PeachScan
+import eu.kanade.tachiyomi.network.GET
+import eu.kanade.tachiyomi.network.asObservable
 import eu.kanade.tachiyomi.network.interceptor.rateLimit
 import eu.kanade.tachiyomi.source.ConfigurableSource
+import eu.kanade.tachiyomi.source.model.FilterList
+import eu.kanade.tachiyomi.source.model.MangasPage
 import eu.kanade.tachiyomi.source.model.Page
 import eu.kanade.tachiyomi.source.model.SChapter
+import eu.kanade.tachiyomi.source.model.SManga
+import eu.kanade.tachiyomi.source.online.HttpSource
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+import okhttp3.Interceptor
 import okhttp3.Response
-import org.jsoup.nodes.Element
+import rx.Observable
 import uy.kohesive.injekt.Injekt
 import uy.kohesive.injekt.api.get
+import uy.kohesive.injekt.injectLazy
+import java.text.SimpleDateFormat
+import java.util.Locale
+import java.util.TimeZone
+import kotlin.getValue
 
-class LuraToon :
-    PeachScan(
-        "Lura Toon",
-        "https://luratoons.com",
-        "pt-BR",
-    ),
-    ConfigurableSource {
+class LuraToon : HttpSource(), ConfigurableSource {
+    override val baseUrl = "https://luratoons.com"
+    override val name = "Lura Toon"
+    override val lang = "pt-BR"
+    override val supportsLatest = true
+
+    private val json: Json by injectLazy()
 
     private val preferences: SharedPreferences by lazy {
         Injekt.get<Application>().getSharedPreferences("source_$id", 0x0000)
     }
 
-    override val client = super.client.newBuilder()
+    override val client = network.cloudflareClient
+        .newBuilder()
+        .addInterceptor(::loggedVerifyInterceptor)
+        .addInterceptor(LuraZipInterceptor()::zipImageInterceptor)
         .rateLimit(3)
         .setRandomUserAgent(
             preferences.getPrefUAType(),
             preferences.getPrefCustomUA(),
         )
         .build()
 
+    override fun chapterListRequest(manga: SManga) = GET("$baseUrl/api/obra${manga.url}", headers)
+    override fun mangaDetailsRequest(manga: SManga) = GET("$baseUrl/api/obra/${manga.url.trimStart('/')}", headers)
```

Comment:
point one of them to the other, also are old urls compatible?

Reply:
I didn't understand the question

Reply:
I didn't make the original comment, but I'm pretty sure it's just for deduplication, since they are so similar:
```kt
override fun chapterListRequest(manga: SManga) = GET("$baseUrl/api/obra/${manga.url.trimStart('/')}", headers)
override fun mangaDetailsRequest(manga: SManga) = chapterListRequest(manga)
// or
override fun mangaDetailsRequest(manga: SManga) = GET("$baseUrl/api/obra/${manga.url.trimStart('/')}", headers)
override fun chapterListRequest(manga: SManga) = mangaDetailsRequest(manga)
```

As for the last question, if the URL changed (i.e. `/manga/123` -> `/comic/some-title-1609520326`), old library entries will be broken. You can bump `versionId` to force migration instead.

If the URL format is exactly the same (i.e. `/manga/123` -> `/manga/123`) existing library entries will continue to work, and `versionId` should be kept.

More of an advanced topic, but if the URL changed in a predictable manner (i.e. `/manga/123` -> `/mangas/item.aspx?id=123`) you could rewrite it to avoid migration. This might cause issues from the browse screen, where entries are shown as being not in the library if the URL is different.

Reply:
> I didn't make the original comment, but I'm pretty sure it's just for deduplication, since they are so similar:
> 
> ```kotlin
> override fun chapterListRequest(manga: SManga) = GET("$baseUrl/api/obra/${manga.url.trimStart('/')}", headers)
> override fun mangaDetailsRequest(manga: SManga) = chapterListRequest(manga)
> // or
> override fun mangaDetailsRequest(manga: SManga) = GET("$baseUrl/api/obra/${manga.url.trimStart('/')}", headers)
> override fun chapterListRequest(manga: SManga) = mangaDetailsRequest(manga)
> ```
> 
> As for the last question, if the URL changed (i.e. `/manga/123` → `/comic/some-title-1609520326`), old library entries will be broken. You can bump `versionId` to force migration instead.
> 
> If the URL format is exactly the same (i.e. `/manga/123` → `/manga/123`) existing library entries will continue to work, and `versionId` should be kept.
> 
> More of an advanced topic, but if the URL changed in a predictable manner (i.e. `/manga/123` → `/mangas/item.aspx?id=123`) you could rewrite it to avoid migration. This might cause issues from the browse screen, where entries are shown as being not in the library if the URL is different.

Regarding deduplication, I understand the intention behind it, but personally, I feel it adds a bit more verbosity to the code. This is just my opinion, but while these functions are essentially the same, they serve different purposes. If, in the future, someone needs to maintain them separately, it could potentially lead to some confusion. However, I understand where you all are coming from and appreciate your approach. I'll go ahead and make the changes as you suggested.

It would be the first case, /manga/123 → /comic/some-title-1609520326. The change is not predictable, and I also don't know which URLs have changed—only some of them did, not all.

Reply:
In that case you should also bump `versionId` to force migration. 

I don't really care for deduplication, it tends to increase coupling, so there's always a tradeoff.

Reply:
> `versionId`para

Ok I will update the versionId